packages/ddl-shim/src/ddl2.ts

```diff
@@ -288,9 +291,14 @@ export interface IView {
 //  DDL  ======================================================================
 export interface Schema {
     version: "0.0.22";
+    createdBy: {
+        name: string;
+        version: string;
+    };
```

Comment:
@drealeed (not 100% if this is a good idea), but whichever app writes the DDL should fill these details in?

Reply:
Might be a good idea to capture each package's version here as an encoded json object. Something like:
```javascript
version_meta = JSON.stringify(btoa({"common":"2.2.4","chart":"2.4.0","marshaller":"2.6.4"}))
```

Reply:
That info or something similar will got into the props - this createdBy will also be used by HIPIE.
FYI - Just catching the @hpcc-js/loader version will give us a definitive version for all the packages (via the loaders package.json).